src/test/java/tests/web/RemoveProductFromCartTest.java

```diff
@@ -0,0 +1,37 @@
+package tests.web;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import pages.CartPage;
+import pages.InventoryPage;
+
+import static constants.ConstantsForProject.EXPECTED_CART_BADGE_COUNT;
+import static constants.ConstantsForProject.SAUCE_LABS_ONESIE;
+
+public class RemoveProductFromCartTest extends AddProductInCartBeforeTest{
+
+    @Test
+    public void testRemoveProductFromCart() {
+
+        CartPage cartPage = new CartPage(getDriver());
+
+        cartPage.removeProductFromCart(SAUCE_LABS_ONESIE);
+        Assert.assertTrue(cartPage.isProductNotInCart(SAUCE_LABS_ONESIE), "Product should be removed from cart");
+    }
+
+    @Test
+    public void testClearCartAndContinueShopping() {
+        InventoryPage inventoryPage = new InventoryPage(getDriver());
+
+        CartPage cartPage = new CartPage(getDriver());
+        Assert.assertEquals(inventoryPage.getHeaderMenuComponent().getCartBadgeText(), EXPECTED_CART_BADGE_COUNT);
```

Comment:
inventoryPage.getHeaderMenuComponent() put it in
HeaderMenuComponent headerMenu = productPage.getHeaderMenuComponent(); and reuse in both asserts